verification/rotateContextHandle.go

```diff
@@ -0,0 +1,78 @@
+// Licensed under the Apache-2.0 license
+
+package verification
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestRotateContextHandle(d TestDPEInstance, c DPEClient, t *testing.T) {
+	testRotateContextHandle(d, c, t, false)
+}
+
+func TestRotateContextHandleSimulation(d TestDPEInstance, c DPEClient, t *testing.T) {
+	testRotateContextHandle(d, c, t, true)
+}
+
+func testRotateContextHandle(d TestDPEInstance, c DPEClient, t *testing.T, simulation bool) {
+	handle := getInitialContextHandle(d, c, t, simulation)
+	defer func() {
+		if simulation {
+			c.DestroyContext(handle, DestroyDescendants)
+		}
+	}()
+
+	// Check handle rotated is non-default when no flags are set
+	handle = testWithoutTargetDefaultFlag(d, c, t, handle)
+
+	// Check handle is rotated to default handle in default context and error is reported in simulated context
+	testWithTargetDefaultFlag(d, c, t, handle, simulation)
+}
+
+// Checks whether the default context handle is rotated to a some other context handle
+func testWithoutTargetDefaultFlag(d TestDPEInstance, c DPEClient, t *testing.T, handle *ContextHandle) *ContextHandle {
```

Comment:
This function has a bit of a side-effect where it rotates the default context. Subsequent tests only succeed because `testWithTargetDefaultFlag` is called after.

I think we should split out the logic for simulation and non-simulation because the tests are checking different things.

For the non-simulation test, we have get the default context, rotate it to be non-default, and then rotate it back. I think the fact that these operations are done in two separate helpers makes the code pretty hard to follow.

Reply:
Yes, this is done, I have added the logic under TestRotateContextHandle() and TestRotateContextHandleSimulation()
I earlier thought to have a function for each case like - what if the flag is set in Default/Sim mode, what if the flag is not set in Default/Sim mode. May that became the source of confusion. Anyway I fixed it now.

Reply:
@jhand2 , let me know if this is fine after addressing the review comments

Reply:
Thanks! Looks good.